### ISSUE TITLE
sentence_eval last token logits

### ISSUE THREAD

**Issue by mhmdsabry** [ericwtodd/function_vectors#4]
In eval_utils.py function sentence_eval (lines 138-189), why are the logits of the last token in sentence (ICL prompt) taken, and not the logits of the last token in the target_completion (which consists of the sentence + target)? I have attached the relevant lines below for reference

```
Line 157: inputs = tokenizer(sentence, return_tensors='pt').to(device)

Line 158: original_pred_idx = len(inputs.input_ids.squeeze()) - 1

Line 170: clean_output = output.logits[:,original_pred_idx,:]

Line 181: clean_output = model(**inputs).logits[:,-1,:]
```

**Comment by ericwtodd** [comment]
The main reason for this is that we are primarily interested in what the model will predict when given the ICL prompt, which is why we use logits at the last token of the ICL sentence. The `target_completion` (sentence+target) prompt is used to compute perplexity, trying to capture how likely it is that the model will predict the same answer as provided in the dataset. The logits at the last token of `target_completion` don't really capture the same thing, as it already contains the answer in the prompt.
